client/src/context/auth/AuthState.js

```diff
@@ -0,0 +1,110 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+import AuthContext from './authContext';
+import authReducer from './authReducer';
+
+import setAuthToken from '../../utils/setAuthToken';
+
+import {
+  REGISTER_SUCCESS,
+  REGISTER_FAIL,
+  USER_LOADED,
+  AUTH_ERROR,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  LOGOUT,
+  CLEAR_ERRORS,
+} from '../types.js';
+
+const AuthState = (props) => {
+  const initialState = {
+    token: localStorage.getItem('token'),
+    isAuthenticated: null,
+    loading: true,
+    error: null,
+    user: null,
+  };
+  const [state, dispatch] = useReducer(authReducer, initialState);
+
+  // Load User
+  const loadUser = async () => {
+    if (localStorage.token) {
+      setAuthToken(localStorage.token);
+    }
+    try {
+      const res = await axios.get('/api/auth');
```

Comment:
this api request will not work if client and server are hosted cross domain.
have a constant with server url and make a request accordingly :)